src/manager.c

```diff
@@ -32,6 +32,49 @@ manager_time(time_t* t)
     return time(t);
 }
 
+FILE*
+manager_fopen(const char* filename, const char* mode)
+{
+    FILE* res = fopen(filename, mode);
+    assert(res != NULL);
+    struct Compartment* comp = manager_find_compartment_by_ddc(cheri_ddc_get()); // TODO
+    void* comp_addr = manager_register_mem_alloc(comp, sizeof(FILE));
```

Comment:
I guess this slowly leaks memory?

Reply:
I think if it does, it'd be the `memcpy` at line 42, but I'm unsure (I assume the `fclose` would take care of that?). The `manager_register_mem_alloc` should not leak memory in the current design, as there is no `mmap` being done within, just updating internal records, and the whole `mmap`'d chunk of compartment scratch memory is `unmap`'d at the end of the program execution.

Reply:
If (say) a program opens and immediately closes a million files, the compartment will run out of memory I think?

Reply:
But the heap is limited either way [1]. Can the files reside out of the compartment if we're enforcing data compartmentalization?

[1] https://github.com/capablevms/CHERI-ELF-comp/blob/master/src/compartment.c#L131

Reply:
Ah, I understand. I would like to avoid `free`ing any stuff, as the memory management design with a bump allocator precludes freeing. So could I add a comment indicating that opening files consumes internal compartment memory due to this design, and a future memory management design would potentially tackle this issue?

Reply:
Apart from the "potentially" bit, fine (I mean future design *must* tackle this at some point ;)).

Reply:
Added in https://github.com/capablevms/CHERI-ELF-comp/pull/4/commits/7ab18fd2849a3f42a6f2281bbeb0d26b1f8bad17.